### PR TITLE
Various: Fix PHP warnings

### PR DESCRIPTION
**Pull request by kraftbj**
Fixes two warnings:

Warning: Illegal offset type in isset or empty in /srv/public/wp-content/plugins/jetpack/modules/sharedaddy/sharing-service.php on line 307

E_WARNING: /home/public_html/wp-content/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-instagram-gallery.php:137 - Invalid argument supplied for foreach()

#### Changes proposed in this Pull Request:

* Validation of variable type before use.

#### Does this pull request change what data or activity we track or use?
n/a

#### Testing instructions:

* For the Sharing one, connected JP, activate Sharing and hit an URL like https://continuing-sacred.jurassic.ninja/2020/07/09/hello-world/?share[0]=twitter&share[1]=facebook&nb=1
(notice the `share[0]` and `share[1]` values.

#### Proposed changelog entry for your changes:
* Bug Fix: Prevent PHP warnings in two instances.

### CODE REVIEW

**Comment by kraftbj** [comment]
One more just came through; standby.

**Comment by kraftbj** [comment]
I'm not thrilled with the third fix.

File: json-endpoints/class.wpcom-json-api-list-posts-v1-1-endpoint.php [hunk 1]
```diff
@@ -382,7 +382,9 @@ function callback( $path = '', $blog_id = 0 ) {
 						if ( ! isset( $return[$key] ) ) {
 							$return[$key] = (object) array();
 						}
-						$return[$key]->next_page = $this->build_page_handle( $last_post, $query );
+							if ( isset( $last_post['ID'] ) ) {
+								$return[ $key ]->next_page = $this->build_page_handle( $last_post, $query );
+							}
```
Comment (jeherve) [inline json-endpoints/class.wpcom-json-api-list-posts-v1-1-endpoint.php:387]: Indenting if off I think, no?

```suggestion
						if ( isset( $last_post['ID'] ) ) {
							$return[ $key ]->next_page = $this->build_page_handle( $last_post, $query );
						}
```

File: json-endpoints/class.wpcom-json-api-list-posts-v1-2-endpoint.php [hunk 2]
```diff
@@ -375,7 +375,10 @@ function callback( $path = '', $blog_id = 0 ) {
 						if ( ! isset( $return[$key] ) ) {
 							$return[$key] = (object) array();
 						}
-						$return[$key]->next_page = $this->build_page_handle( $last_post, $query );
+
+							if ( isset( $last_post['ID'] ) ) {
+								$return[ $key ]->next_page = $this->build_page_handle( $last_post, $query );
+							}
```
Comment (jeherve) [inline json-endpoints/class.wpcom-json-api-list-posts-v1-2-endpoint.php:381]:
```suggestion
						if ( isset( $last_post['ID'] ) ) {
							$return[ $key ]->next_page = $this->build_page_handle( $last_post, $query );
						}
```

Comment (kraftbj) [reply json-endpoints/class.wpcom-json-api-list-posts-v1-2-endpoint.php:381]: php-changed flagged those lines as needed the extra tab. I didn't run phpcs on the whole file, but I can do that.

Comment (kraftbj) [reply json-endpoints/class.wpcom-json-api-list-posts-v1-2-endpoint.php:398]: 92e5123

**Comment by kraftbj** [comment]
r210269-wpcom

**Comment by jeherve** [comment]
Cherry-picked to `branch-8.7` in 30ae815f44e003af1af86da95689d5bce88c8f6e

**Pull request merged**